readthedocs/proxito/views/serve.py

```diff
@@ -148,7 +148,7 @@ class ServeDocs(SettingsOverrideObject):
     _default_class = ServeDocsBase
 
 
-class ServeError404Base(ServeRedirectMixin, View):
+class ServeError404Base(ServeRedirectMixin, ServeDocsMixin, View):
```

Comment:
We are inheriting a lot of functionality (`allow_user`, `_serve_docs`, etc) that it's not required in this view. We are rendering the template from django instead serving a doc.

I think I'd prefer if we can have a small mixin with the method `get_version_from_host` instead, which will make this easier to read.

Reply:
Are you doing this small refactor?

Reply:
I'm not sure that having even more mixins is really a good solution either. I feel that having too many mixins makes the code hard to read, so I'm not convinced that it's better.

Reply:
In particular, all of the doc-serving code needs `get_version_from_host`, so it would require adding a new mixin to *all* the classes in this function, and I don't see a lot of value in that.